### PR TITLE
Install os-autoinst-distri-opensuse-deps in worker container

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/97751

### CODE REVIEW

**Review by Martchus — APPROVED** [review]
I also noticed that problem when testing https://github.com/os-autoinst/openQA/pull/4160.

It doesn't seem to be the right approach to tailor our container to a specific test distribution and rather provide a way (e.g. instructions in the documentation) for installing additional dependencies as needed. Obviously, we couldn't accept such a PR for any possible test distribution and the arbitrary dependencies they might need.

However, for the sake of providing a good out-of-the-box experience for openSUSE tests I'm approving the PR nevertheless. There's no harm done for other test distributions.

**Review by perlpunk — APPROVED** [review]

**Review by okurz — CHANGES_REQUESTED** [review]
Absolutely not. This is not a "openSUSE openQA worker"

**Comment by asdil12** [comment]
New approach:

The test distribution will get a script into the git root that will install the dependencies. The container entrypoint script will call such a script in the test distribution folder if present.

See also: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13244

File: container/worker/run_openqa_worker.sh [hunk 1]
```diff
@@ -24,4 +24,14 @@ fi
 qemu-system-x86_64 -S &
 kill $!
 
+if [[ -e /var/lib/openqa/share/tests ]] ; then
+  pushd /var/lib/openqa/share/tests
+    for dir in $(ls) ; do
```
Comment (okurz) [inline container/worker/run_openqa_worker.sh:29]: `$(ls)` should be equivalent to `*`, isn't it?

File: container/worker/run_openqa_worker.sh [hunk 2]
```diff
@@ -24,4 +24,14 @@ fi
...
+      pushd "$dir"
+      test -x install_deps.sh && ./install_deps.sh
+      popd
+    done
+  popd
+fi
```
Comment (Martchus) [inline container/worker/run_openqa_worker.sh:35]: The code is bad for multiple reasons, e.g. it breaks if white spaces are used and produces noisy `pushd`/`popd` output. I'd just use `find`:
```suggestion
find '/hdd/openqa-devel/openqa/share/tests' -maxdepth 2 -type f -executable -name 'install_deps.sh' -exec {} \;
```

**Review by Martchus — CHANGES_REQUESTED** [review]
I suppose this is acceptable although I must say that downloading/installing packages every time when starting a worker is not ideal. I'm only requesting changes because the code should be simplifies (see inline comment).

File: container/worker/run_openqa_worker.sh (hunk 1 above)
Comment (Martchus) [reply container/worker/run_openqa_worker.sh:29]: `*` would be better because it doesn't break when using white spaces. However, I'd prefer using `find`.

**Review by Martchus — APPROVED** [review]

**Review by okurz — APPROVED** [review]

**Pull request merged**